libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/ConfigurationBotFrameworkAuthentication.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Rest;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core
+{
+    /// <summary>
+    /// Creates a <see cref="BotFrameworkAuthentication"/> instance from configuration.
+    /// </summary>
+    public class ConfigurationBotFrameworkAuthentication : BotFrameworkAuthentication
+    {
+        private readonly BotFrameworkAuthentication _inner;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConfigurationBotFrameworkAuthentication"/> class.
+        /// </summary>
+        /// <param name="configuration">An IConfiguration instance.</param>
+        /// <param name="credentialsFactory">An IServiceClientCredentialsFactory instance.</param>
+        /// <param name="authConfiguration">An AuthenticationConfiguration instance.</param>
+        /// <param name="httpClient">A custom HttpClient to use.</param>
+        /// <param name="logger">The ILOgger instance to use.</param>
+        public ConfigurationBotFrameworkAuthentication(IConfiguration configuration, ServiceClientCredentialsFactory credentialsFactory = null, AuthenticationConfiguration authConfiguration = null, HttpClient httpClient = null, ILogger logger = null)
+        {
+            var channelService = configuration.GetSection("ChannelService")?.Value;
```

Comment:
Should we add validations and throw exceptions if any of the required config elements is missing?

Reply:
I think we could - it would make it more friendly - when we understand this a little better perhaps. I've created an issue to track this: https://github.com/microsoft/botbuilder-dotnet/issues/4672

(as a general theme better error reporting on misconfiguration would probably be very useful given how locked down some of these cloud environments might be)